modules/tinymce/src/plugins/visualchars/main/ts/core/Bindings.ts

```diff
@@ -5,16 +5,15 @@
  * For commercial licenses see https://www.tiny.cloud/
  */
 
+import { Cell } from '@ephox/katamari';
 import Editor from 'tinymce/core/api/Editor';
 import * as Settings from '../api/Settings';
 import * as Actions from './Actions';
 
-const setup = (editor: Editor, toggleState) => {
+const setup = (editor: Editor, toggleState: Cell<boolean>) => {
   editor.on('init', () => {
-    // should be false when enabled, so toggling will change it to true
-    const valueForToggling = !Settings.isEnabledByDefault(editor);
-    toggleState.set(valueForToggling);
-    Actions.toggleVisualChars(editor, toggleState);
+    toggleState.set(Settings.isEnabledByDefault(editor));
+    Actions.applyVisualChars(editor, toggleState);
```

Comment:
Hmm... I'm not sure this is safe still. As the editor will try and place the initial selection before `init` and running this code will potentially cause the selection to be lost.

As a general rule, any direct dom manipulation needs to have a bookmark stored, so the selection isn't lost as the dom is manipulated.

Reply:
Discussed via slack and this is a very large edge case and isn't likely to be much of an issue. There are also many worse things with the visualchars plugin and we want to rewrite it at some stage. As such we should just add a comment or something to that effect, in case it does come up again as a potential issue.

What's required to trigger this case:
- Auto focus or manually focus the editor on init.
- First bit of text must be a nbsp
- Integrator has manually set the selection before `init`